src/Products/Viewer/Controllers/ViewerApiController.cs

```diff
@@ -203,225 +167,119 @@ public HttpResponseMessage RotateDocumentPages(PostedDataEntity postedData)
-                        }
-                    }
-                }
-                else
-                {
-                    using (WebClient client = new WebClient())
-                    {
-                        // get file name from the URL
-                        Uri uri = new Uri(url);
-                        string fileName = Path.GetFileName(uri.LocalPath);
-                        if (rewrite)
-                        {
-                            // Get the complete file path
-                            fileSavePath = Path.Combine(documentStoragePath, fileName);
-                        }
-                        else
-                        {
-                            fileSavePath = Resources.GetFreeFileName(documentStoragePath, fileName);
-                        }
-                        // Download the Web resource and save it into the current filesystem folder.
-                        client.DownloadFile(url, fileSavePath);
-                    }
-                }
-                UploadedDocumentEntity uploadedDocument = new UploadedDocumentEntity();
-                uploadedDocument.guid = fileSavePath;
-                return Request.CreateResponse(HttpStatusCode.OK, uploadedDocument);
-            }
-            catch (System.Exception ex)
-            {
-                // set exception message
-                return Request.CreateResponse(HttpStatusCode.InternalServerError, new Resources().GenerateException(ex));
-            }
-        }
+            // get/set parameters
+            string documentGuid = postedData.guid;
 
-        [HttpPost]
-        [Route("loadThumbnails")]
-        public LoadDocumentEntity loadThumbnails(PostedDataEntity loadDocumentRequest)
-        {
-            return LoadDocument(loadDocumentRequest, true);
-        }
+            dynamic viewInfo;
 
-        [HttpPost]
-        [Route("loadPrint")]
-        public HttpResponseMessage loadPrint(PostedDataEntity loadDocumentRequest)
-        {
-            try
-            {
-                LoadDocumentEntity loadPrintDocument = LoadDocument(loadDocumentRequest, true);
-                // return document description
-                return Request.CreateResponse(HttpStatusCode.OK, loadPrintDocument);
+            string pageFilePathFormat;
+            ViewerSettings settings = GetViewerSettings(documentGuid, out pageFilePathFormat);
 
-            }
-            catch (System.Exception ex)
+            using (GroupDocs.Viewer.Viewer viewer = new GroupDocs.Viewer.Viewer(documentGuid, settings))
             {
-                // set exception message
-                return Request.CreateResponse(HttpStatusCode.InternalServerError, new Resources().GenerateException(ex, loadDocumentRequest.password));
+                HtmlViewOptions options = HtmlViewOptions.ForEmbeddedResources(pageFilePathFormat);
+                viewInfo = viewer.GetViewInfo(ViewInfoOptions.ForHtmlView());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this useless assignment to local variable 'viewInfo'.](https://app.codacy.com/gh/groupdocs-viewer/GroupDocs.Viewer-for-.NET-MVC/pullRequest?prid=5242614)